lib/persistence/settings.rb

```diff
@@ -111,7 +111,12 @@ def settings(prefix)
     end
 
     def threshold
-      config[:health_check_threshold_in_minutes] || DEFAULT_HEALTHCHECK_THRESHOLD
+      threshold_param = config[:health_check_threshold_in_minutes]
+      if threshold_param && threshold_param != ""
+        return threshold_param
+      end
+      
+      DEFAULT_HEALTHCHECK_THRESHOLD
```

Comment:
```
def threshold
  threshold_param = config[:health_check_threshold_in_minutes].to_s
  return threshold_param unless threshold_param.empty?
  DEFAULT_HEALTHCHECK_THRESHOLD
end
```

"to_s" works even on nil values, and "empty?" will be true if it was a nil value ("to_s" would convert it to an empty string)